lib/mauth/client/local_authenticator.rb

```diff
@@ -81,22 +82,21 @@ def signature_valid_v2!(object)
           query_string: euresource_escape(original_query_string.to_s)
         )
 
-        actual = actual_string_to_sign(object)
-
-        unless expected_no_reencoding == actual ||
-           expected_euresource_style_reencoding == actual ||
-           expected_for_percent_reencoding == actual
+        unless verify_signature!(object, expected_no_reencoding) ||
+           verify_signature!(object, expected_euresource_style_reencoding) ||
+           verify_signature!(object, expected_for_percent_reencoding)
           msg = "Signature verification failed for #{object.class}"
           log_inauthentic(object, msg)
           raise InauthenticError, msg
         end
       end
 
-      def actual_string_to_sign(object)
+      def verify_signature!(object, expected_str_to_sign)
         pubkey = OpenSSL::PKey::RSA.new(retrieve_public_key(object.signature_app_uuid))
```

Comment:
Potentially this method is called 3 times, can the initialization of pubkey and digest moved out ?